pytest.skip causes tests to fail on cleanup

## Description

In a test that uses pytest-jupyter fixture, if I call `pytest.skip()`, the test will fail during cleanup with a timeout error! See e.g. https://github.com/jupyterlab/jupyterlab_server/pull/324

## Context

```python
___________ ERROR at teardown of test_contents_manager_preferred_dir ___________

io_loop = <tornado.platform.asyncio.AsyncIOMainLoop object at 0x7f7ad4b25c50>
http_server_port = (<socket.socket [closed] fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6>, 36149)
jp_web_app = <jupyter_server.serverapp.ServerWebApplication object at 0x7f7ad3ea1a50>

    @pytest.fixture
    def http_server(io_loop, http_server_port, jp_web_app):
        """Start a tornado HTTP server that listens on all available interfaces."""
    
        async def get_server():
            server = tornado.httpserver.HTTPServer(jp_web_app)
            server.add_socket(http_server_port[0])
            return server
    
        server = io_loop.run_sync(get_server)
        yield server
        server.stop()
    
        if hasattr(server, "close_all_connections"):
>           io_loop.run_sync(server.close_all_connections)

../../../.local/share/hatch/env/virtual/jupyterlab-server/DCUu2YYM/cov/lib/python3.7/site-packages/pytest_jupyter/jupyter_server.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tornado.platform.asyncio.AsyncIOMainLoop object at 0x7f7ad4b25c50>
func = <bound method HTTPServer.close_all_connections of <tornado.httpserver.HTTPServer object at 0x7f7ad3eaedd0>>
timeout = None

    def run_sync(self, func: Callable, timeout: Optional[float] = None) -> Any:
        """Starts the `IOLoop`, runs the given function, and stops the loop.
    
        The function must return either an awaitable object or
        ``None``. If the function returns an awaitable object, the
        `IOLoop` will run until the awaitable is resolved (and
        `run_sync()` will return the awaitable's result). If it raises
        an exception, the `IOLoop` will stop and the exception will be
        re-raised to the caller.
    
        The keyword-only argument ``timeout`` may be used to set
        a maximum duration for the function.  If the timeout expires,
        a `asyncio.TimeoutError` is raised.
    
        This method is useful to allow asynchronous calls in a
        ``main()`` function::
    
            async def main():
                # do stuff...
    
            if __name__ == '__main__':
                IOLoop.current().run_sync(main)
    
        .. versionchanged:: 4.3
           Returning a non-``None``, non-awaitable value is now an error.
    
        .. versionchanged:: 5.0
           If a timeout occurs, the ``func`` coroutine will be cancelled.
    
        .. versionchanged:: 6.2
           ``tornado.util.TimeoutError`` is now an alias to ``asyncio.TimeoutError``.
        """
        future_cell = [None]  # type: List[Optional[Future]]
    
        def run() -> None:
            try:
                result = func()
                if result is not None:
                    from tornado.gen import convert_yielded
    
                    result = convert_yielded(result)
            except Exception:
                fut = Future()  # type: Future[Any]
                future_cell[0] = fut
                future_set_exc_info(fut, sys.exc_info())
            else:
                if is_future(result):
                    future_cell[0] = result
                else:
                    fut = Future()
                    future_cell[0] = fut
                    fut.set_result(result)
            assert future_cell[0] is not None
            self.add_future(future_cell[0], lambda future: self.stop())
    
        self.add_callback(run)
        if timeout is not None:
    
            def timeout_callback() -> None:
                # If we can cancel the future, do so and wait on it. If not,
                # Just stop the loop and return with the task still pending.
                # (If we neither cancel nor wait for the task, a warning
                # will be logged).
                assert future_cell[0] is not None
                if not future_cell[0].cancel():
                    self.stop()
    
            timeout_handle = self.add_timeout(self.time() + timeout, timeout_callback)
        self.start()
        if timeout is not None:
            self.remove_timeout(timeout_handle)
        assert future_cell[0] is not None
        if future_cell[0].cancelled() or not future_cell[0].done():
>           raise TimeoutError("Operation timed out after %s seconds" % timeout)
E           concurrent.futures._base.TimeoutError: Operation timed out after None seconds

../../../.local/share/hatch/env/virtual/jupyterlab-server/DCUu2YYM/cov/lib/python3.7/site-packages/tornado/ioloop.py:528: TimeoutError
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -30,7 +30,12 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+      - name: Run the tests on pypy
+        if: ${{ startsWith(matrix.python-version, 'pypy') }}
+        run: |
+          hatch run test:nowarn || hatch run test:nowarn --lf
       - name: Run the tests
+        if: ${{ !startsWith(matrix.python-version, 'pypy') }}
         run: hatch run cov:test || hatch run test:test --lf
       - name: Coverage
         run: |

--- a/pytest_jupyter/jupyter_server.py
+++ b/pytest_jupyter/jupyter_server.py
@@ -67,7 +67,10 @@ def http_server(io_loop, http_server_port, jp_web_app):
     server.stop()
 
     if hasattr(server, "close_all_connections"):
-        io_loop.run_sync(server.close_all_connections)
+        try:
+            io_loop.run_sync(server.close_all_connections)
+        except TimeoutError:
+            pass
 
     http_server_port[0].close()
 

--- a/tests/test_jupyter_server.py
+++ b/tests/test_jupyter_server.py
@@ -2,6 +2,7 @@ import json
 import os
 from unittest.mock import MagicMock
 
+import pytest
 from jupyter_server.auth import Authorizer
 from jupyter_server.serverapp import ServerApp
 from tornado.websocket import WebSocketHandler
@@ -63,3 +64,7 @@ def test_template_dir(jp_template_dir):
 
 def test_extension_environ(jp_extension_environ):
     pass
+
+
+def test_skip(jp_serverapp):
+    pytest.skip("Forcing a skip")

--- a/tests/test_jupyter_server.py
+++ b/tests/test_jupyter_server.py
@@ -14,6 +14,10 @@ async def test_serverapp(jp_serverapp):
     assert isinstance(jp_serverapp, ServerApp)
 
 
+def test_skip(jp_serverapp):
+    pytest.skip("Forcing a skip")
+
+
 async def test_get_api_spec(jp_fetch):
     response = await jp_fetch("api", "spec.yaml", method="GET")
     assert response.code == 200
@@ -64,7 +68,3 @@ def test_template_dir(jp_template_dir):
 
 def test_extension_environ(jp_extension_environ):
     pass
-
-
-def test_skip(jp_serverapp):
-    pytest.skip("Forcing a skip")

